From 8fca5e646d2ee42f0b0b27d89a0032e9be4db081 Mon Sep 17 00:00:00 2001
From: Chao Liu <chao.liu@processmission.com>
Date: Tue, 14 Jul 2026 16:00:00 +0800
Subject: [PATCH] mmc: spacemit: allow fixed-rate input clocks

A fixed-rate clock has no set_rate operation and returns -ENOSYS.  The
requested rate is already provided in that case, so allow the controller to
continue probing.  Preserve failures from clock providers which implement
set_rate.

Signed-off-by: Chao Liu <chao.liu@processmission.com>
---
 drivers/mmc/k1x_sdhci.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/k1x_sdhci.c b/drivers/mmc/k1x_sdhci.c
index b1916a8a1d..0c4b9748ef 100644
--- a/drivers/mmc/k1x_sdhci.c
+++ b/drivers/mmc/k1x_sdhci.c
@@ -617,7 +617,7 @@ static inline int spacemit_sdhci_get_clocks(struct udevice *dev)
 	}
 
 	ret = clk_set_rate(&clk, host->max_clk);
-	if (ret) {
+	if (ret && ret != -ENOSYS) {
 		dev_err(dev, "Failed to set io clk: %d\n", ret);
 		return ret;
 	}
-- 
2.50.1

